src/java/org/apache/cassandra/cql3/functions/FormatFcts.java

```diff
@@ -0,0 +1,426 @@
+     * <p>
+     * Supported units are: B, KiB, MiB, GiB
+     * <p>
+     * Supported column types on which this function is possible to be applied:
+     * <pre>INT, TINYINT, SMALLINT, BIGINT, VARINT, ASCII, TEXT</pre>
+     * For ASCII and TEXT types, text of such column has to be a non-negative number.
+     * <p>
+     *
+     * The conversion of negative values is not supported.
+     */
+    public static class FormatBytesFct extends NativeScalarFunction
+    {
+        private static final String FUNCTION_NAME = "format_bytes";
+
+        private FormatBytesFct(AbstractType<?>... argsTypes)
+        {
+            super(FUNCTION_NAME, UTF8Type.instance, argsTypes);
+        }
+
+        @Override
+        public ByteBuffer execute(Arguments arguments) throws InvalidRequestException
+        {
+            if (arguments.get(0) == null)
+                return null;
+
+            if (arguments.containsNulls())
+                throw new InvalidRequestException("none of the arguments may be null");
+
+            long value = getValue(arguments);
+
+            if (value < 0)
+                throw new InvalidRequestException("value must be non-negative");
+
+            DataStorageUnit sourceUnit;
+            DataStorageUnit targetUnit;
+
+            if (arguments.size() == 1)
+            {
+                sourceUnit = BYTES;
+
+                if (value > FileUtils.ONE_GIB)
+                    targetUnit = GIBIBYTES;
+                else if (value > FileUtils.ONE_MIB)
+                    targetUnit = MEBIBYTES;
+                else if (value > FileUtils.ONE_KIB)
+                    targetUnit = KIBIBYTES;
+                else
+                    targetUnit = BYTES;
+            }
+            else if (arguments.size() == 2)
+            {
+                sourceUnit = BYTES;
+                targetUnit = validateUnit(arguments.get(1));
+            }
+            else
+            {
+                sourceUnit = validateUnit(arguments.get(1));
+                targetUnit = validateUnit(arguments.get(2));
+            }
+
+            long convertedValue = convertValue(value, sourceUnit, targetUnit);
+
+            return UTF8Type.instance.fromString(convertedValue + " " + targetUnit.getSymbol());
+        }
+
+        private long convertValue(long valueToConvert, DataStorageUnit sourceUnit, DataStorageUnit targetUnit)
+        {
+            if (targetUnit == BYTES)
+                return sourceUnit.toBytes(valueToConvert);
+            else if (targetUnit == KIBIBYTES)
+                return sourceUnit.toKibibytes(valueToConvert);
+            else if (targetUnit == MEBIBYTES)
+                return sourceUnit.toMebibytes(valueToConvert);
+            else if (targetUnit == GIBIBYTES)
+                return sourceUnit.toGibibytes(valueToConvert);
+            else
+                throw new InvalidRequestException("unsupported target unit " + targetUnit);
```

Comment:
Same as before: This makes me wonder, should we move the validate unit calls inside this method to avoid having to call them from the main execute method?